hyp3_metadata/rtc.py

```diff
@@ -0,0 +1,113 @@
+            elif isinstance(output, Path):
+                files.append(output)
+
+        return files
+
+    def create_readme(self) -> Path:
+        reference_file = self.payload['product_dir'] / f'{self.payload["product_dir"].name}_{self.payload["polarizations"][0]}.tif'
+
+        return self.create_metadata_file(self.payload, 'rtc/readme.md.txt.j2', reference_file, out_ext='README.md.txt', strip_ext=True, strip_pol=True
+        )
+
+    def create_product_xmls(self) -> List[Path]:
+        payload = deepcopy(self.payload)
+
+        output_files = []
+        for pol in payload['polarizations']:
+            payload['pol'] = pol
+            reference_file = payload['product_dir'] / f'{payload["product_dir"].name}_{pol}.tif'
+
+            output_files.append(
+                self.create_metadata_file(payload, 'rtc/product.xml.j2', reference_file)
+            )
+        return output_files
+
+    def create_dem_xml(self) -> Path:
+        reference_file = self.payload['product_dir'] / f'{self.payload["product_dir"].name}_dem.tif'
+
+        dem_template_id = get_dem_template_id(self.payload['dem_name'])
+        if dem_template_id is not None:
+            return self.create_metadata_file(self.payload, f'dem/dem-{dem_template_id}.xml.j2', reference_file)
+
+    def create_browse_xmls(self) -> List[Path]:
+        reference_file = self.payload['product_dir'] / f'{self.payload["product_dir"].name}.png'
+
+        output_files = [
+            self.create_metadata_file(self.payload, 'browse/browse-greyscale.xml.j2', reference_file),
+        ]
+
+        rgb_file = self.payload['product_dir'] / f'{self.payload["product_dir"].name}_rgb.png'
+        if rgb_file.exists():
+            output_files.append(
+                self.create_metadata_file(self.payload, 'browse/browse-color.xml.j2', rgb_file)
+            )
+        return output_files
+
+    def create_inc_map_xml(self) -> Path:
+        reference_file = self.payload['product_dir'] / f'{self.payload["product_dir"].name}_inc_map.tif'
+        return self.create_metadata_file(self.payload, 'rtc/inc_map.xml.j2', reference_file)
+
+    def create_ls_map_xml(self) -> Path:
+        reference_file = self.payload['product_dir'] / f'{self.payload["product_dir"].name}_ls_map.tif'
+        return self.create_metadata_file(self.payload, 'rtc/ls_map.xml.j2', reference_file)
+
+    def create_area_xml(self) -> Path:
+        reference_file = self.payload['product_dir'] / f'{self.payload["product_dir"].name}_area.tif'
+        return self.create_metadata_file(self.payload, 'rtc/area.xml.j2', reference_file)
+
+    def create_rgb_xml(self) -> Path:
+        reference_file = self.payload['product_dir'] / f'{self.payload["product_dir"].name}_rgb.tif'
+        return self.create_metadata_file(self.payload, 'rtc/rgb.xml.j2', reference_file)
+
+    @classmethod
+    def create_metadata_file(cls, payload: dict, template: str, reference_file: Path, out_ext: str = 'xml',
```

Comment:
I want this function to be generalized, but I don't think it can be, once we start implementing insar docs we will have a better idea of what can be reused.